sunbeam-python/sunbeam/commands/juju.py

```diff
@@ -358,7 +427,7 @@ def run(self, status: Optional["Status"] = None) -> Result:
         :return:
         """
         try:
-            result = self.add_cloud(self.cloud, self.definition)
+            result = self.add_cloud(self.cloud, self.definition, self.controller)
             if not result:
                 return Result(ResultType.FAILED, "Unable to create cloud")
         except subprocess.CalledProcessError as e:
```

Comment:
The error handling is not great for users, this step is a step I think will often fail because users might miss the grant super user part

Here's what it looks like:

```
                juju.py:434
                    Traceback (most recent call last):                                                                                                                                                                                                   
                      File "/snap/openstack/x1/lib/python3.10/site-packages/sunbeam/commands/juju.py", line 430, in run                                                                                                                                  
                        result = self.add_cloud(self.cloud, self.definition, self.controller)                                                                                                                                                            
                      File "/snap/openstack/x1/lib/python3.10/site-packages/sunbeam/commands/juju.py", line 224, in add_cloud                                                                                                                            
                        process = subprocess.run(cmd, capture_output=True, text=True, check=True)                                                                                                                                                        
                      File "/usr/lib/python3.10/subprocess.py", line 526, in run                                                                                                                                                                         
                        raise CalledProcessError(retcode, process.args,                                                                                                                                                                                  
                    subprocess.CalledProcessError: Command '['/snap/openstack/x1/juju/bin/juju', 'add-cloud', 'sunbeam', '--file', '/tmp/tmprgjg889_', '--client', '--controller', 'infinity-book', '--force']' returned non-zero exit status            
                    1.                                                                                                                                                                                                                                   
           WARNING  Cloud "sunbeam" successfully added to your local client.                                                                                                                                                                  juju.py:435
                    You will need to add a credential for this cloud (`juju add-credential sunbeam`)                                                                                                                                                     
                    before you can use it to bootstrap a controller (`juju bootstrap sunbeam`) or                                                                                                                                                        
                    to create a model (`juju add-model <your model name> sunbeam`).                                                                                                                                                                      
                                                                                                                                                                                                                                                         
                    Could not upload cloud to a controller: permission denied (unauthorized access)                                                                                                                                                      
                                                                                                                                                                                                                                                         
           DEBUG    Finished running step 'Add Cloud'. Result: ResultType.FAILED                                                                                                                                                            common.py:275
Error: Command '['/snap/openstack/x1/juju/bin/juju', 'add-cloud', 'sunbeam', '--file', '/tmp/tmprgjg889_', '--client', '--controller', 'infinity-book', '--force']' returned non-zero exit status 1.
```

Reply:
Re-running this step after having granted the user failed with:

```
⠦ Adding cloud to Juju ... [10:12:43] ERROR    Error adding cloud to Juju                                                                                                                                                                                                juju.py:434
                    Traceback (most recent call last):                                                                                                                                                                                                   
                      File "/snap/openstack/x1/lib/python3.10/site-packages/sunbeam/commands/juju.py", line 430, in run                                                                                                                                  
                        result = self.add_cloud(self.cloud, self.definition, self.controller)                                                                                                                                                            
                      File "/snap/openstack/x1/lib/python3.10/site-packages/sunbeam/commands/juju.py", line 224, in add_cloud                                                                                                                            
                        process = subprocess.run(cmd, capture_output=True, text=True, check=True)                                                                                                                                                        
                      File "/usr/lib/python3.10/subprocess.py", line 526, in run                                                                                                                                                                         
                        raise CalledProcessError(retcode, process.args,                                                                                                                                                                                  
                    subprocess.CalledProcessError: Command '['/snap/openstack/x1/juju/bin/juju', 'add-cloud', 'sunbeam', '--file', '/tmp/tmp7qgqw8br', '--client', '--controller', 'infinity-book', '--force']' returned non-zero exit status            
                    1.                                                                                                                                                                                                                                   
           WARNING                                                                                                                                                                                                                            juju.py:435
                    Cloud "sunbeam" added to controller "infinity-book".                                                                                                                                                                                 
                    WARNING loading credentials: credentials for cloud sunbeam not found                                                                                                                                                                 
                    To upload a credential to the controller for cloud "sunbeam", use                                                                                                                                                                    
                    * 'add-model' with --credential option or                                                                                                                                                                                            
                    * 'add-credential -c sunbeam'.                                                                                                                                                                                                       
                    ERROR use `update-cloud sunbeam --client` to override known definition: local cloud "sunbeam" already exists                                                                                                                         
                                                                                                                                                                                                                                                         
           DEBUG    Finished running step 'Add Cloud'. Result: ResultType.FAILED                                                                                                                                                            common.py:275
Error: Command '['/snap/openstack/x1/juju/bin/juju', 'add-cloud', 'sunbeam', '--file', '/tmp/tmp7qgqw8br', '--client', '--controller', 'infinity-book', '--force']' returned non-zero exit status 1.
```